mathics/builtin/specialfns/hypergeom.py

```diff
@@ -41,95 +158,103 @@ class HypergeometricPFQ(MPMathFunction):
 
     Result is symbollicaly simplified by default:
     >> HypergeometricPFQ[{3}, {2}, 1]
-     = HypergeometricPFQ[{3}, {2}, 1]
+     = 3 E / 2
+
     unless a numerical evaluation is explicitly requested:
     >> HypergeometricPFQ[{3}, {2}, 1] // N
      = 4.07742
+
     >> HypergeometricPFQ[{3}, {2}, 1.]
      = 4.07742
 
+    >> Plot[HypergeometricPFQ[{1, 1}, {3, 3, 3}, x], {x, -30, 30}]
+     = -Graphics-
+
+    >> HypergeometricPFQ[{1, 1, 2}, {3, 3}, z]
+     = -4 PolyLog[2, z] / z ^ 2 + 4 Log[1 - z] / z ^ 2 - 4 Log[1 - z] / z + 8 / z
+
     The following special cases are handled:
     >> HypergeometricPFQ[{}, {}, z]
-     = 1
+     = E ^ z
     >> HypergeometricPFQ[{0}, {b}, z]
      = 1
-    >> Hypergeometric1F1[b, b, z]
-     = E ^ z
+
+     >> HypergeometricPFQ[{1, 1, 3}, {2, 2}, x]
+      = -Log[1 - x] / (2 x) - 1 / (-2 + 2 x)
+
+    'HypergeometricPFQ' evaluates to a polynomial if any of the parameters $a_k$ is a non-positive integer:
+    >> HypergeometricPFQ[{-2, a}, {b}, x]
+     = (-2 a x (1 + b) + a x ^ 2 (1 + a) + b (1 + b)) / (b (1 + b))
+
+    Value at origin:
+    >> HypergeometricPFQ[{a1, b2, a3}, {b1, b2, b3}, 0]
+     = 1
     """
 
     attributes = A_NUMERIC_FUNCTION | A_PROTECTED | A_READ_PROTECTED
     mpmath_name = "hyper"
     nargs = {3}
-    rules = {
-        "HypergeometricPFQ[{}, {}, z_]": "1",
-        "HypergeometricPFQ[{0}, b_, z_]": "1",
-        "HypergeometricPFQ[b_, b_, z_]": "Exp[z]",
-    }
+
     summary_text = "compute the generalized hypergeometric function"
     sympy_name = "hyper"
 
     def eval(self, a, b, z, evaluation: Evaluation):
         "HypergeometricPFQ[a_, b_, z_]"
-        try:
-            a_sympy = [e.to_sympy() for e in a]
-            b_sympy = [e.to_sympy() for e in b]
-            result_sympy = tracing.run_sympy(
-                sympy.hyper, a_sympy, b_sympy, z.to_sympy()
+
+        # FIXME: a lot more checking could be done here.
+        if not (isinstance(a, ListExpression)):
+            evaluation.message(
+                "HypergeometricPQF",
+                "hdiv",
+                Expression(Symbol("Hypergeometric"), a, b, z),
```

Comment:
I will be revising this to use  `Pattern[]` in the signature docstring as described in https://github.com/Mathics3/mathics-development-guide/pull/44 to reduce the clumsiness of having to recreate an expression that already exists.